source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -22,6 +22,8 @@ ProxyFilterConfig::ProxyFilterConfig(
 
 ProxyFilter::ProxyFilter(ProxyFilterConfigSharedPtr config) : config_(std::move(config)) {}
 
+ProxyFilter::~ProxyFilter() { circuit_breaker_.reset(); }
```

Comment:
you dont need to manually reset this, it should get destroyed as part of the default object teardown

Reply:
done